tests/test-backend-ops.cpp

```diff
@@ -36,12 +36,18 @@
 #include <thread>
 #include <vector>
 
+static size_t get_n_threads(const int64_t ne) {
+    const size_t max_threads_hw = std::max(std::thread::hardware_concurrency()/2, (unsigned int)1);
+    const size_t max_threads_ne = (ne + 1024 - 1) / 1024;
+    return std::min(max_threads_hw, max_threads_ne);
+}
+
 static void init_tensor_uniform(ggml_tensor * tensor, float min = -1.0f, float max = 1.0f) {
     size_t nels = ggml_nelements(tensor);
     std::vector<float> data(nels);
     {
         // parallel initialization
-        static const size_t n_threads = std::thread::hardware_concurrency();
+        static const size_t n_threads = get_n_threads(ggml_nelements(tensor));
```

Comment:
I don't think that it makes sense to initialize this from the current tensor, since this is a one-time RNG initialization. It should be done with the maximum number of threads that may be used.